docs/spec/light/api.md

```diff
@@ -0,0 +1,391 @@
+
+url: /keys/create
+
+Functionality: Create a new key.
+
+Parameter:
+| Parameter | Type   | Default | Required | Description      |
+| --------- | ------ | ------- | -------- | ---------------- |
+| name      | string | null    | true     | name of key      |
+| password  | string | null    | true     | password of key  |
+
+Returns on success:
+
+```json
+{
+    "rest api": "2.0",
+    "code":200,
+    "error": "",
+    "result": {
+        "seed":crime carpet recycle erase simple prepare moral dentist fee cause pitch trigger when velvet animal abandon
+    }
+}
+```
+
+Returns on failure:
+
+```json
+{
+    "rest api": "2.0",
+    "code":500,
+    "error": "Could not create new key.",
+    "result": {}
+}
+```
+
+### /keys/{name} - GET
+
+url: /keys/{name}
+
+Functionality: Get the information for the specified key.
+
+Returns on success:
+
+```json
+{
+    "rest api": "2.0",
+    "code":200,
+    "error": "",
+    "result": {
+        "name": "test",
+            "address": "cosmosaccaddr1thlqhjqw78zvcy0ua4ldj9gnazqzavyw4eske2",
+            "pub_key": "cosmosaccpub1zcjduc3qyx6hlf825jcnj39adpkaxjer95q7yvy25yhfj3dmqy2ctev0rxmse9cuak"
+    }
+}
+```
+
+Returns on failure:
+
+```json
+{
+    "rest api": "2.0",
+    "code":500,
+    "error": "Could not find information on the specified key.",
+    "result": {}
+}
+```
+
+### /keys/{name} - PUT
+
+url: /keys/{name}
+
+Functionality: Change the encryption password for the specified key.
+
+Parameters:
+| Parameter       | Type   | Default | Required | Description     |
+| --------------- | ------ | ------- | -------- | --------------- |
+| old_password    | string | null    | true     | old password    |
+| new_password    | string | null    | true     | new password    |
+
+Returns on success:
+
+```json
+{
+    "rest api": "2.0",
+    "code":200,
+    "error": "",
+    "result": {}
+}
+```
+
+Returns on failure:
+
+```json
+{
+    "rest api": "2.0",
+    "code":500,
+    "error": "Could not update the specified key.",
+    "result": {}
+}
+```
+
+### /keys/{name} - DELETE
+
+url: /keys/{name}
+
+Functionality: Delete the specified key.
+
+Parameters:
+| Parameter | Type   | Default | Required | Description      |
+| --------- | ------ | ------- | -------- | ---------------- |
+| password  | string | null    | true     | password of key  |
+
+Returns on success:
+
+```json
+{
+    "rest api": "2.0",
+    "code":200,
+    "error": "",
+    "result": {}
+}
+```
+
+Returns on failure:
+
+```json
+{
+    "rest api": "2.0",
+    "code":500,
+    "error": "Could not delete the specified key.",
+    "result": {}
+}
+```
+
+## ICS20 - TokenAPI
+
+The TokenAPI exposes all functionality needed to query account balances and send transactions.
+
+### /bank/balance/{account} - GET
+
+url: /bank/balance/{account}
+
+Functionality: Query the specified account.
+
+Returns on success:
+
+```json
+{
+    "rest api": "2.0",
+    "code":200,
+    "error": "",
+    "result": {
+        "atom": 1000,
```

Comment:
Amino v0.10 follows proto3 where integers are quoted as strings

Reply:
Does this apply to all integers? Will the error codes change as well?

Reply:
I think we should stick with integers here though, since it seems to be the standard for json. This API doesn't necessarily have to comply with amino encoding.